### PR TITLE
Fix code quality in Model

### PR DESCRIPTION
**Pull request by migfoo02**
1. Changed the `MESSAGE_CONSTRAINTS` message in `Role` and 'Company' to be more specific. Fixed date time format for `Deadline`.
2. Simplified `if` statement for `Deadline`
3. Updated regex in Company to only accept alphanumeric characters for the first character.

### CODE REVIEW

File: src/main/java/seedu/application/model/job/Company.java [hunk 1]
```diff
@@ -57,7 +59,7 @@ public boolean equals(Object other) {
         }
 
         Company otherCompany = (Company) other;
-        return name.equals(otherCompany.name);
+        return name.equalsIgnoreCase(otherCompany.name);
```
Comment (evanyan13) [inline src/main/java/seedu/application/model/job/Company.java:62]: Thank you! Looks like a better implementation.

File: src/main/java/seedu/application/model/job/Deadline.java [hunk 2]
```diff
@@ -15,7 +15,7 @@ public class Deadline {
 
     public static final String MESSAGE_CONSTRAINTS =
         "Deadline should be in valid DateTime format: "
-            + "MMMM dd yyyy HHmm\n"
+            + "MMM dd yyyy HHmm\n"
```
Comment (evanyan13) [inline src/main/java/seedu/application/model/job/Deadline.java:18]: Thank you for the sharp spotting of the error

**Review by evanyan13 — APPROVED** [review]

File: src/main/java/seedu/application/model/job/Role.java [hunk 3]
```diff
@@ -12,7 +12,9 @@ public class Role {
 
     public static final String ROLE_FIND_SPECIFIER = "-r";
     public static final String MESSAGE_CONSTRAINTS =
-            "Role descriptions should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Role descriptions should adhere to the following constraints:\n"
+                    + "1. Only contain alphanumeric characters and spaces\n"
```
Comment (JeremyYong128) [inline src/main/java/seedu/application/model/job/Role.java:16]: Nice use of numbered list to specify constraints!

**Review by JeremyYong128 — APPROVED** [review]

File: src/main/java/seedu/application/model/job/Company.java [hunk 4]
```diff
@@ -11,13 +11,15 @@
 public class Company {
     public static final String COMPANY_FIND_SPECIFIER = "-c";
     public static final String MESSAGE_CONSTRAINTS =
-            "Company names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Company names should adhere to the following constraints:\n"
+                    + "1. Start with an alphanumeric character\n"
+                    + "2. Should not be blank";
```
Comment (peiran18) [inline src/main/java/seedu/application/model/job/Company.java:16]: This looks much clearer!

**Pull request merged**